### PR TITLE
chore: update development environment script

### PR DESCRIPTION
**Pull request by salasberryfin**
**What this PR does / why we need it**:

Tilt environment was failing to run `rancher-turtles-controller` because some of the pre-install operations performed when using `helm install` were not applied. To compensate for this, the script that creates the development environment is now updated to install `cluster-api-operator` (together with `cert-manager`) and default to `rancher:v2.8.1`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

As a follow-up, we should also add a test case that covers the development environment #339.

**Checklist**:

- [x] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Review by furkatgofurov7 — COMMENTED** [review]
Two questions:

1. have you verified all controllers are shown in the Tilt UI (I am guessing Tiltfile might need to be adjusted to visualize those)
2. Wondering why CAPI Operator has not been installed in this script from the beginning?

**Comment by salasberryfin** [comment]
@furkatgofurov7 right now only Rancher Turtles is shown in Tilt's UI. My Tilt knowledge is quite limited but I understand this is expected as we're not watching/building any controller other than Turtles and they're simply installed when configuring the environment. Should we be building this as is done in core CAPI's Tilt setup?  
Don't know about CAPI Operator, but Rancher Turtles was failing as it was looking for the resource.

File: scripts/turtles-dev.sh [hunk 1]
```diff
@@ -15,48 +15,47 @@
...
 
 helm install rancher rancher-stable/rancher \
-    --namespace cattle-system \
-    --create-namespace \
-    --set bootstrapPassword=rancheradmin \
-    --set replicas=1 \
-    --set hostname="$RANCHER_HOSTNAME" \
-    --set 'extraEnv[0].name=CATTLE_FEATURES' \
-    --set 'extraEnv[0].value=embedded-cluster-api=false' \
-    --set global.cattle.psp.enabled=false \
```
[the remark this answers is not in the document]
Comment (salasberryfin) [reply scripts/turtles-dev.sh:55]: Looks like this is not required when using `v2.7.5` and above: https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-references/helm-chart-options#advanced-options

**Comment by furkatgofurov7** [comment]
> My Tilt knowledge is quite limited but I understand this is expected as we're not watching/building any controller other than Turtles and they're simply installed when configuring the environment. Should we be building this as is done in core CAPI's Tilt setup?

No, at least with not this PR maybe, but overall it is possible to add other controllers to the ALL.controllers list in the Tiltfile and get their logs shown in the Tilt UI which would be easier to debug.

File: scripts/turtles-dev.sh [hunk 2]
```diff
@@ -15,48 +15,47 @@
...
-if [ -z "$RANCHER_HOSTNAME" ]
-then
-    echo "You must pass a rancher host name"
-    exit 1
+if [ -z "$RANCHER_HOSTNAME" ]; then
+	echo "You must pass a rancher host name"
+	exit 1
 fi
 
+RANCHER_VERSION=${RANCHER_VERSOIN:-v2.8.1}
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply scripts/turtles-dev.sh:23]: Can you address this nit, otherwise PR LGTM

**Comment by salasberryfin** [comment]
Thanks @furkatgofurov7. Having logs from other controllers would definitely be more useful, I'll take a look at how to add this to the Tilt configuration and update this PR.

**Comment by richardcase** [comment]
> Thanks @furkatgofurov7. Having logs from other controllers would definitely be more useful, I'll take a look at how to add this to the Tilt configuration and update this PR.

@salasberryfin @furkatgofurov7 - i think this only originally showed the turtles controller. So i'd propose lets get it working with that only and then think about adding other controllers in the future. wdyt?

**Review by richardcase — COMMENTED** [review]
Looks good to me, just the comments that @furkatgofurov7 mentioned already.

**Comment by salasberryfin** [comment]
Resolved reviews and moved from WIP.

**Comment by furkatgofurov7** [comment]
Just remembered, since this is updated we need to adapt notes accordingly in docs repo: https://docs.rancher-turtles.com/docs/contributing/development#what-happens-when-you-run-make-dev-env

**Review by furkatgofurov7 — APPROVED** [review]

**Review by richardcase — APPROVED** [review]

**Comment by richardcase** [comment]
Thanks for this @salasberryfin .

**Pull request merged**